### PR TITLE
Clean up/enable StringBuilder.append tests

### PR DESCRIPTION
**Pull request by allredj**
Some StringBuilder tests were marked as FUTURE and caused by https://github.com/diffblue/test-gen/issues/434.
This seems now fixed, so I moved the tests to the main StringBuilder.append tests and enabled them,  but I added a `--max-nondet-string-length 1000` option. Not sure whether these tests should always pass without the option.
@romainbrenguier do you think we can remove it (the tests don't fail locally without the option, but I'm worried they might at a later time)?

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java [hunk 1]
```diff
@@ -1,10 +1,16 @@
 public class test
 {
-   public static void main(String[] args)
-   {
-      StringBuilder sb = new StringBuilder(args[0]);
-      sb.append("Z");
-      String s = sb.toString();
-      assert(s.equals("fg"));
-   }
+    public static void nondet(String stringArg, int intArg)
```
Comment (romainbrenguier) [inline jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java]: I don't see where intArg is used

Comment (romainbrenguier) [inline jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java]: :pick: the name suggest there should also be a `det` test

**Review by romainbrenguier — APPROVED** [review]

File: jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java (hunk 1 above)
Comment (allredj) [reply jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java]: It's not. Removed.

Comment (allredj) [reply jbmc/regression/jbmc-strings/StringBuilderAppend03/test.java]: The tests for deterministic inputs are in the StringBuilderAppend[01] folders.

File: jbmc/regression/jbmc-strings/StringBuilderAppend03/test1.desc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CORE
+test.class
+--function test.nondet --max-nondet-string-length 1000 --cp `../../../../scripts/format_classpath.sh . ../../../lib/java-models-library/target/core-models.jar` --property "java::test.nondet:(Ljava/lang/String;)V.assertion.1"
+^EXIT=10$
+^SIGNAL=0$
+\[.*assertion\.1\] .* line 9 .* FAILURE
```
Comment (peterschrammel) [inline jbmc/regression/jbmc-strings/StringBuilderAppend03/test1.desc:6]: What about the other two assertions?

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 62ceb1bd82bf1d9ce6f340ca566e06d1020f1ea1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117650646

File: jbmc/regression/jbmc-strings/StringBuilderAppend03/test1.desc (hunk 2 above)
Comment (allredj) [reply jbmc/regression/jbmc-strings/StringBuilderAppend03/test1.desc:6]: I'm testing per-property, so the other assertions are checked in the two other desc files. Is that what you mean?

**Pull request merged**